src/grt/src/GlobalRouter.cpp

```diff
@@ -221,17 +221,23 @@ void GlobalRouter::saveCongestion()
   }
   std::ofstream out(congestion_file_name_);
 
-  std::vector<std::pair<GSegment, TileCongestion>> congestionGridsV,
-      congestionGridsH;
+  std::vector<CongestionInformation> congestionGridsV, congestionGridsH;
   fastroute_->getCongestionGrid(congestionGridsV, congestionGridsH);
   for (auto& it : congestionGridsH) {
+    const auto& [seg, tile, srcs] = it;
     out << "violation type: Horizontal congestion\n";
-    const int capacity = it.second.first;
-    const int usage = it.second.second;
-    out << "\tsrcs: \n";
+    const int capacity = tile.first;
+    const int usage = tile.second;
+    if (!srcs.empty()) {
```

Comment:
I don't think we need this check anymore as we can't have overflow with no sources